### PR TITLE
Add tracking events to the Spotlight component

### LINKED ISSUE #61266: Add tracking events to the Spotlight component
https://github.com/Automattic/wp-calypso/pull/61185#discussion_r810112392

Since we are creating this new shiny component, I think it would worth also adding some track events for users who have clicked through it. This way we can measure the performance (click through rate) and run A/B tests later.

An example can be found here
https://github.com/Automattic/wp-calypso/blob/0178a4ae1aa2031071dc846cfc0344b353fd233e/client/blocks/announcement-modal/index.jsx#L52-L56

### PR DESCRIPTION
**Pull request by Rolilink**
#### Changes proposed in this Pull Request

* Track events to the Spotlight component.

#### Testing instructions
* Show tracking in your browser: PCYsg-cae-p2
* Start with a site on a < Business plan
* Go to /plugins/< site id >?flags=marketplace-spotlight
* Go through the flow and check that the `calypso_marketplace_spotlight_click` track event is logged with the site domain, type, plugin slug, and plugin id after the confirm step.

Related to #61266

### CODE REVIEW

File: client/my-sites/plugins/plugins-browser-list/test/index.jsx [hunk 1]
```diff
@@ -6,6 +6,10 @@ jest.mock( 'i18n-calypso', () => ( {
 	localize: ( c ) => c,
 	translate: ( s ) => s,
 } ) );
+jest.mock( 'react-redux', () => ( {
```
Comment (allilevine) [inline client/my-sites/plugins/plugins-browser-list/test/index.jsx:9]: @lupus2k Why is this needed? To dispatch the track event from the test?

File: client/my-sites/plugins/plugins-browser-list/test/index.jsx [hunk 2]
```diff
@@ -20,6 +24,7 @@ const plugins = [
 
 const props = {
 	plugins,
+	variant: PluginsBrowserListVariant.Fixed,
```
Comment (allilevine) [inline client/my-sites/plugins/plugins-browser-list/test/index.jsx:27]: It doesn't look like we're passing in the other variants as props, why is it needed here?

File: client/my-sites/plugins/plugins-browser-list/index.jsx [hunk 3]
```diff
@@ -87,6 +91,17 @@ const PluginsBrowserList = ( {
 		}
 	};
 
+	const spotlightOnClick = () => {
+		dispatch(
+			recordTracksEvent( 'calypso_marketplace_spotlight_click', {
```
Comment (allilevine) [inline client/my-sites/plugins/plugins-browser-list/index.jsx:96]: Do we also need to record impressions, to measure click-through rate? Like `calypso_banner_cta_impression`

File: client/my-sites/plugins/plugins-browser-list/test/index.jsx (hunk 1 above)
Comment (lupus2k) [reply client/my-sites/plugins/plugins-browser-list/test/index.jsx:9]: Exactly, useDispatch comes from react-redux

File: client/my-sites/plugins/plugins-browser-list/test/index.jsx (hunk 2 above)
Comment (lupus2k) [reply client/my-sites/plugins/plugins-browser-list/test/index.jsx:27]: The prop `variant` is marked as required in `PluginsBrowserList` and should be one of the values [listed here](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plugins/plugins-browser-list/types.ts)

File: client/my-sites/plugins/plugins-browser-list/index.jsx (hunk 3 above)
Comment (lupus2k) [reply client/my-sites/plugins/plugins-browser-list/index.jsx:96]: It is a good idea, I think we have to create a new task for that, what do you think?

File: client/my-sites/plugins/plugins-browser-list/index.jsx [hunk 4]
```diff
@@ -87,6 +91,17 @@ const PluginsBrowserList = ( {
...
+				type: 'plugin',
+				slug: spotlightPlugin.slug,
+				id: spotlightPlugin.id,
+				site: site,
+			} )
+		);
+		page( `/plugins/${ spotlightPlugin.slug }/${ site }` );
```
Comment (lsl) [inline client/my-sites/plugins/plugins-browser-list/index.jsx]:
```suggestion
		page( `/plugins/${ spotlightPlugin.slug }/${ site || '' }` );
```
`site` can be null if there is no site selected:

Screenshot_2022-03-01_14-49-42

**Review by lsl — APPROVED** [review]

File: client/my-sites/plugins/plugins-browser-list/index.jsx (hunk 3 above)
Comment (allilevine) [reply client/my-sites/plugins/plugins-browser-list/index.jsx:96]: Sounds good!
